AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -271,14 +272,23 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
         }
     }
 
+    func onItemChangeDuringSync(mutationEvent: MutationEvent) {
+        serialQueue.async {
```

Comment:
If so, should we add it on all `serialQueue.async` invocations? 

I see there are a lot in this file.

Reply:
good catch! the serialQueue is dispatching the closure on global queue. Could it cause the retain cycles?

Reply:
> good catch! the serialQueue is dispatching the closure on global queue. Could it cause the retain cycles?
Because the receiver (`DataStoreObserverQueryOperation`) holds a strong reference to `serialQueue`, there is a temporary retain cycle while the block (who retains `self`) is waiting to be executed. So, because this is temporary, it's probably not a big deal - however, my opinion is that it's good practice worth exercising so that you do it when it does become necessary.

Reply:
👍  will update the code for all the async closuer on queue